api/tests/Behat/bootstrap/v2/ClientManagement/ClientManagementTrait.php

```diff
@@ -369,4 +369,58 @@ public function theClientDoesNotHaveANamedDeputyAssociatedWithThem()
         $this->em->persist($client);
         $this->em->flush();
     }
+
+    /**
+     * @When I attempt to unarchived the client
+     */
+    public function iAttemptToUnarchivedTheClient()
+    {
+        $this->assertInteractingWithUserIsSet();
+
+        try {
+            $this->clickLink('Un-archive client');
+            $this->iAmOnAdminClientUnarchivedPage();
+            $this->clickLink('Return to client dashboard');
+        } catch (\Throwable $e) {
+            // This step is used as part of testing the unarchive button isnt here so swallow errors and assert on following step
+        }
+    }
+
+    /**
+     * @When an org deputy has an archived client
+     */
+    public function theDeputyHasAnArchivedClient()
+    {
+        $clientId = $this->interactingWithUserDetails->getClientId();
+
+        /** @var Client $client */
+        $client = $this->em->getRepository(Client::class)->find($clientId);
+
+        $client->setArchivedAt(new DateTime('yesterday'));
+
+        $this->em->persist($client);
+        $this->em->flush();
+    }
+
+    /**
+     * @Then the client should be unarchived
+     */
+    public function theClientShouldBeUnarchived()
+    {
+        $this->assertInteractingWithUserIsSet();
+        $this->iAmOnAdminClientDetailsPage();
```

Comment:
Are we actually asserting on anything here? If we're displaying a flash message or something similar we could assert on that instead.

Reply:
Ah OK gotcha 👍